analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,415 @@
+  inner_join(
+    consensus_seg_cytoband_status_df,
+    by = c("biospecimen_id" = "Kids_First_Biospecimen_ID", "cytoband")
+  )
+```
+
+## Define most focal units
+
+### Determine chromosome arm status
+
+```{r}
+# Use our custom function to make the status calls
+arm_status_count <- status_majority_caller(
+  combined_status_df,
+  "chromosome_arm",
+  "status", 
+  percent_threshold = percent_threshold
+)
+
+# Display table
+arm_status_count %>%
+  group_by(dominant_status)
+```
+
+These are the chromosome arms that have not been defined as gain or loss -- we want to define their cytoband/gene level status
+
+```{r}
+# Let's get a vector of the neutral arms
+neutral_arms <- arm_status_count %>%
+  filter(dominant_status == "neutral") %>% 
+  dplyr::pull("chromosome_arm")
+```
+
+### Determine cytoband status
+
+We want to include cytoband and gene level calls for chromosome arms that have not been defined as a gain or loss.
+
+```{r}
+# Filter the annotated CN data to include only neutral chromosome arms
+neutral_status_arm_df <- combined_status_df %>%
+  filter(chromosome_arm %in% neutral_arms)
+```
+
+Making cytoband-level majority calls. 
+
+```{r}
+if (!(running_in_ci)) {
+# Now count the cytoband level calls (for each status call) and define
+# the cytoband as that status if more than 50% of the total counts are
+# for that particular status
+cytoband_status_count <- status_majority_caller(
+  neutral_status_arm_df,
+  "cytoband",
+  "dominant_cytoband_status",
+  percent_threshold = percent_threshold
+)
+
+# Display table
+cytoband_status_count
+}
+```
+
+### Determine gene-level status
+
+```{r}
+if (!(running_in_ci)) {
+  # These are the cytobands that have not been defined as gain or loss --
+  # we want to define their gene level status
+  neutral_cytobands <- cytoband_status_count %>%
+    filter(dominant_status %in% c("unstable", "neutral")) %>%
+    dplyr::pull("cytoband")
+
+  # Filter the annotated CN data to include only these cytobands
+  neutral_status_cytoband_df <- combined_status_df %>%
+    filter(cytoband %in% neutral_cytobands)
+  
+  # Now count the gene level calls (for each status call) and define
+  # the gene as that status if more than 50% of the total counts are
+  # for that particular status
+  gene_status_count <- status_majority_caller(neutral_status_cytoband_df,
+                                              "gene_symbol",
+                                              "status", 
+                                              percent_threshold = percent_threshold)
+  
+  # Display table
+  gene_status_count
+}
+```
+
+## Plot calls
+
+Plot the final dominant status call on the x-axis and the percent of each status on the y-axis.
+
+### Plot chromosome arm status calls
+
+```{r}
+# Run `plot_dominant_status_calls` function for the chromosome arm calls
+plot_dominant_status_calls(
+  arm_status_count,
+  "chromosome_arm"
+)
+```
+
+### Plot cytoband status calls
+
+```{r fig.width = 10}
```

Comment:
Don't need this `fig.width` anymore.

Reply:
I added this because these particular plots have the two additional labels on the x-axis so without the `fig.width` argument the labels start to overlap and become hard to read.

Reply:
Oh Sorry, I misread this chunk. It still has a plot. Nevermind!